File: src/main/java/com/brett/lec06/Lec06Main.java
package com.brett.lec06;

import java.util.Arrays;
import java.util.List;

public class Lec06Main {
    public static void main(String[] args) {
        whilePrint();
    }

    private static void listPrint(){
        List<Long> numbers = Arrays.asList(1L, 2L, 3L);
        for(long number : numbers){
            System.out.println(number);
        }
    }

    private static void forPrint(){
        for(int i=1;i<=3;i++){
            System.out.println(i);
        }
    }

    private static void forDownPrint(){
        for(int i=3;i>=1;i--){
            System.out.println(i);
        }
    }

    private static void for2StepUpPrint(){
        for(int i=1;i<=5;i+=2){
            System.out.println(i);
        }
    }

    private static void whilePrint(){
        int i = 1;
        while(i<=3){
            System.out.println(i);
            i++;
        }
    }
}

File: src/main/kotlin/com/brett/lec06/Lec06Main.kt
package com.brett.lec06

fun main(){
//    listPrint()
//    forDownPrint()
    for2StepUpPrint()
}

fun listPrint(){
    /**
     * Arrays.asList -> listOf
     */
    val numbers = listOf(1L, 2L, 3L)
    for (number in numbers) {
        println(number)
    }
}

fun forPrint(){
    for (i in 1..3) {
        println(i)
    }
}

fun forDownPrint(){
    for (i in 3 downTo 1) {
        println(i)
    }
}

fun for2StepUpPrint(){
    for (i in 1..5 step 2){
        println(i)
    }
}

fun whilePrint(){
    var i = 1
    while (i <= 3) {
        println(i)
        i++
    }
}